src/proof/alethe/alethe_printer.cpp

```diff
@@ -0,0 +1,265 @@
+      return it->second;
+    }
+
+    // Otherwise, print anchor
+    Trace("alethe-printer")
+        << "... print anchor " << pfn->getResult() << " " << arule << " "
+        << " / " << args << std::endl;
+    out << "(anchor :step " << current_prefix << "t" << current_step_id;
+
+    // Append index of anchor to prefix so that all steps in the subproof use it
+    current_prefix.append("t" + std::to_string(current_step_id));
+
+    // Reset the current step id s.t. the numbering inside the subproof starts
+    // with 1
+    current_step_id = 1;
+
+    // If the subproof is a bind the arguments need to be printed as
+    // assignments, i.e. args=[(= v0 v1)] is printed as (:= (v0 Int) v1).
+    if (arule == AletheRule::ANCHOR_BIND)
+    {
+      out << " :args (";
+      for (size_t j = 3, size = args.size(); j < size; j++)
+      {
+        out << "(:= (" << args[j][0] << " " << args[j][0].getType() << ") "
+            << args[j][1] << ")";
+        if (j != args.size() - 1)
+        {
+          out << " ";
+        }
+      }
+      out << ")";
+    }
+    // In all other cases there are no arguments
+    out << ")\n";
+
+    // If the subproof is a genuine subproof the arguments are printed as
+    // assumptions. To be able to discharge the assumptions afterwards we need
+    // to store them.
+    if (arule == AletheRule::ANCHOR_SUBPROOF)
+    {
+      for (size_t i = 3, size = args.size(); i < size; i++)
+      {
+        std::string assumption_name =
+            current_prefix + ".a" + std::to_string(i - 3);
+        Trace("alethe-printer")
+            << "... print assumption " << args[i] << std::endl;
+        out << "(assume " << assumption_name << " " << args[i] << ")\n";
+        assumptions[args[i]] = assumption_name;
+        current_assumptions.push_back(assumption_name);
+      }
+    }
+  }
+
+  // Assumptions are printed at the anchor and therefore have to be in the list
+  // of assumptions when an assume is reached.
+  else if (arule == AletheRule::ASSUME)
+  {
+    Trace("alethe-printer")
+        << "... reached assumption " << pfn->getResult() << " " << arule << " "
+        << " / " << args << " " << std::endl;
+
+    auto it = assumptions.find(args[2]);
+    if (it != assumptions.end())
+    {
+      Trace("alethe-printer") << "... found assumption in list "
+                              << ": " << args[2] << "/" << assumptions
+                              << "     " << it->second << std::endl;
+      return it->second;
+    }
+
+    Trace("alethe-printer") << "... printing failed! Encountered assumption "
+                               "that has not been printed! "
+                            << args[2] << "/" << assumptions << std::endl;
+    return "";
+  }
+
+  // Print children
+  std::vector<std::string> child_prefixes;
+  std::string prefix = current_prefix;
+  if (current_prefix != "" && current_prefix.back() != '.')
+  {
+    prefix.append(".");
+  }
```

Comment:
I'm not sure if I missed something but couldn't you add the "." already when you update the current prefix?

Reply:
You can but you have to delete the last "." then if you end the subproof. I implemented this now.